python/plugins/processing/algs/qgis/Heatmap.py

```diff
@@ -170,7 +204,7 @@ def processAlgorithm(self, parameters, context, feedback):
         kde = QgsKernelDensityEstimation(kde_params, output, output_format)
 
         if kde.prepare() != QgsKernelDensityEstimation.Success:
-            raise GeoAlgorithmExecutionException(
+            raise QgsProcessingException(
```

Comment:
So it is better to use `GeoAlgorithmExecutionException` or switch to `QgsProcessingException` to reduce (a little) code to maintain?